internal/pkg/skuba/kubernetes/resources.go

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2019 SUSE LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package kubernetes
+
+import (
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+// DoesResourceExist check the given error from a client-go API call and returns
+// whether the resource exists. If no error was given, it will return true, if
+// the given error was a `IsNotFound` it will return false; otherwise it will
+// return an error
+func DoesResourceExist(err error) (bool, error) {
+	switch {
+	case err == nil:
+		return true, nil
+	case apierrors.IsNotFound(err):
+		return false, nil
+	}
```

Comment:
Nitpicking: what's the benefit of using a `switch` statement here instead of mere `if` statements?

Reply:
Thought it was more clear, nothing special. If you think it's more clear with `if` I can change it.

Reply:
I think it's more clear as `if`, but I don't mind. It's really a nitpick